### PR TITLE
Implementation of API

### PR DESCRIPTION
**Pull request by Mekuseo**
- Make use of the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as your backend
- Use either [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) or [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make API calls
- Fetch your books from Bookstore API on when the component has rendered with useEffect
- Refactor your add book and remove book features to persist your changes in the server
- Modify your reducer logic to include HTTP requests to the Bookstore API

### CODE REVIEW

File: src/components/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      return action.payload;
+    default:
+      return state;
+  }
+};
+
+// Action Creators
+const getBooks = () => async (dispatch) => {
+  const allBooks = [];
+  const response = await axios.get(api);
```
Comment (topeogunleye) [inline src/components/redux/books/books.js:9]: - [x] As required, I kindly suggest that you make use of `createAsyncThunk` to fetch data from api.

## EXAMPLE

```
export const fetchBooksData = createAsyncThunk('fetchBooksData', async () => {
  const response = await fetch(api);
  const data = await response.json();
  const booksList = [];
  Object.keys(data).forEach((id) => {
    booksList.push({
      item_id: id,
      title: data[id][0].title,
      author: data[id][0].author,
      category: data[id][0].category,
    });
  });
  return booksList;
});
```


The above code is just an example and may not work directly for your code.

File: src/components/redux/books/booksSlice.js [hunk 2]
```diff
@@ -1,61 +0,0 @@
-/* eslint-disable no-param-reassign */
```
Comment (topeogunleye) [inline src/components/redux/books/booksSlice.js:1]: - [x] I don't know why you deleted this file, but `booksSlice` is here and it is a requirement for your fetch method in `redux/books.js` to update the state of `booksSlice`

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @Mekuseo, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Try-Again

> ## (Highlights) Good Points: 💪 
> * Good commit messages. :heavy_check_mark:
> * No linter errors. :heavy_check_mark:
> * Followed Github flow :heavy_check_mark:
> * Professional README file. :heavy_check_mark:

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] It will be great to not have your redux folder nested within your components folder as they're supposed to be siblings folders.

image


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/redux/books/books.js (hunk 1 above)
Comment (Mekuseo) [reply src/components/redux/books/books.js:9]: Hi Reviewer, 
I have added createAsyncThunk as you suggested.

File: src/components/redux/books/booksSlice.js (hunk 2 above)
Comment (Mekuseo) [reply src/components/redux/books/booksSlice.js:1]: I deleted booksSlice and replaced it with books.js component.

**Review by AnselemOdims — APPROVED** [review]
### Hi  @Mekuseo 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/redux/books/books.js (hunk 1 above)
Comment (topeogunleye) [reply src/components/redux/books/books.js:9]: > Hi Reviewer, I have added createAsyncThunk as you suggested.

@Mekuseo, that's great

**Pull request merged**